engine/evd.mli

```diff
@@ -98,6 +106,10 @@ type evar_info = {
   (** Boolean mask over {!evar_hyps}. Should have the same length.
       When filtered out, the corresponding variable is not allowed to occur
       in the solution *)
+  evar_abstract_arguments : Abstraction.t;
+  (** Boolean information over {!evar_hyps}, telling if an hypothesis instance
+      can be immitated or should stay abstract in HO unification problems
+      and inversion (see [second_order_matching_with_args] for its use). *)
```

Comment:
Am I understanding well that this "abstraction" thing lets you chose a solution for
```
H |- ?e[H] === f(H)
```
among:
- `H |- ?e := fun _ => f(H)`
- ` H |- ?e := fun H1 => f(H1)`
In which case the boolean corresponding to H is supposed to be true/false?

Reply:
Exactly. If the argument is marked abstract then the abstraction solution `(fun H1 => f (H1))` will be chosen, if possible.

Reply:
I think you should clarify the comment, maybe copy pasting the example up there.

In particular it is still not clear to me what `true` means.
In the example above you could just say "when the bool for H is true".

An alternative is to use a variant instead of the boolean, so that one would have `Imitate` or `Abstract` and then the comment would not be needed.

Reply:
I went for the Imitate and Abstract variants here.